java/service_instance.go

```diff
@@ -205,399 +187,560 @@ const (
+	State ServiceInstanceStatus `json:"state"`
+	// Oracle WebLogic Server software version
+	Version string `json:"version"`
+	// Groups details about WLS VM instances by host name. Each VM instance is a JSON object element.
+	VMInstances VMInstances `json:"vmInstances"`
+}
+
+type Clusters struct {
+	ClusterID    int                    `json:"clusterId"`
+	ClusterName  string                 `json:"clusterName"`
+	ClusterType  string                 `json:"clusterType"`
+	CreationDate string                 `json:"creationDate"`
+	PaaSServers  map[string]PaaSServers `json:"paasServers"`
+	// Identifies service specific cluster and server information. Details include:
+	// Cluster type - APPLICATION_CLUSTER or CACHING_CLUSTER
+	// Shape - Compute shape used by nodes of this cluster
+	// Whether this cluster is accessible from the public network (external value is true if accessible)
+	Profile string `json:"profile"`
+}
+
+type PaaSServers struct {
+	// Attribute details of a specific server.
+	Attributes PaaSAttributes `json:"attributes"`
+}
+
+type PaaSAttributes struct {
+	// One or more Managed Server JVM arguments separated by a space.
+	AdditionalJVMArgs string `json:"additional_jvm_args"`
+	// Name of the Coherence cluster (the system-level CoherenceClusterSystemResource) that the WLS cluster
+	// (application or caching) is associated with. Default value is DataGridConfig.
+	CCSR string `json:"ccsr"`
+	// Cluster that this Managed Server is a member of.
+	Cluster string `json:"cluster"`
+	// Java heap size of the Managed Server JVM.
+	HeapSize string `json:"heap_size"`
+	// Initial heap size
+	HeapStart string `json:"heap_start"`
+	// Maximum Permanent Generation (PermGen) space in Java heap memory for a Managed Server JVM.
+	MaxPermSize string `json:"max_perm_size"`
+	// Initial Permanent Generation (PermGen) space in Java heap memory for a Managed Server JVM.
+	PermSize string `json:"perm_size"`
+	// Port number
+	Port string `json:"port"`
+	// Server role
+	Role string `json:"role"`
+	// Server type
+	ServerType string `json:"server_type"`
+	// SSL port number
+	SSLPort string `json:"ssl_port"`
+	// Cluster template for the server. For example: ExampleI_cluster_Template, DataGridServer-Template
+	Template string `json:"template"`
+}
+
+type WLSAttributes struct {
+	// Attributes of the Fusion Middleware (Upper Stack) product to be installed or installed on the service instance.
+	UpperStackProductName AttributeInfo `json:"upperStackProductName"`
+}
+
+type OTDAttributes struct {
+	AdminPort                    AttributeInfo `json:"ADMIN_PORT"`
+	ListenerPort                 AttributeInfo `json:"LISTENER_PORT"`
+	ListenerPortEnabled          AttributeInfo `json:"LISTENER_PORT_ENABLED"`
+	PrivilgedListenerPort        AttributeInfo `json:"PRIV_LISTENER_PORT"`
+	PrivilegedSecureListenerPort AttributeInfo `json:"PRIV_SECURE_LISTENER_PORT"`
+	SecuredListenerPort          AttributeInfo `json:"SECURE_LISTENER_PORT"`
+}
+
+type Hosts struct {
```

Comment:
These wrapper types (`Hosts`, `UserHosts`, `VMInstances`, `CurrentOperation`, etc.) are kind of annoying, you could probably just custom marshal to flatten your structs, but not a big deal.

Reply:
You're right it's annoying but I'd rather keep it this way for the initial release. We can refactor it away if it becomes an issue